File: src/main/java/eu/qwsome/sql/condition/In.java
package eu.qwsome.sql.condition;


import java.util.Arrays;

public class In implements Condition {

  private final ValueHolder field;
  private final ValueHolder[] values;

  In(final ValueHolder field, final ValueHolder... values){
    this.field = field;
    this.values = values;
  }

  @Override
  public CharSequence get() {
    final StringBuilder builder = new StringBuilder();
    appendTo(builder);

    return builder;
  }

  @Override
  public ValueConstructor getValues() {
    final ValueConstructor valueConstructor = new ValueConstructor().add(this.field);
    Arrays.stream(this.values).forEach(valueConstructor::add);
    return valueConstructor;
  }

  @Override
  public void appendTo(StringBuilder builder) {
    builder.append(this.field.getSql())
      .append(" IN ")
      .append("( ");

    int i = 0;
    for (; i < this.values.length - 1; i++) {
      builder.append(this.values[i].getSql())
        .append(", ");
    }

    builder.append(this.values[i].getSql())
      .append(" )");
  }
}

File: src/main/kotlin/eu/qwsome/sql/condition/In.kt
/*
 * @author Lukáš Kvídera
 */
package eu.qwsome.sql.condition

class In(
    private val field: ValueHolder,
    private vararg val values: ValueHolder,
) : Condition {

    override fun getValues(): ValueConstructor {
        val valueConstructor = ValueConstructor().add(field)
        values.forEach { valueConstructor.add(it) }
        return valueConstructor
    }

    override fun appendTo(builder: StringBuilder) {
        builder.append(field.getSql())
            .append(" IN ( ")

        values.forEachIndexed { index, value ->
            if (index > 0) {
                builder.append(", ")
            }
            builder.append(value.getSql())
        }

        builder.append(" )")
    }
}
